src/main/java/frc/robot/subsystems/UpBeat.java

```diff
@@ -54,38 +57,61 @@ public UpBeat() {
         bottomMotor.burnFlash();
     }
 
+    private final static class Constants{
+        private static final double stop = 0.00;
+        private static final double reverse = -1000.00;
+        private static final double shoot = 4000.00;
+        private static final double amp = 1000.00;
+    }
+
+    @AutoLogOutput(key = "upBeat/speed")
+    private double getSpeed() {
+        return bottomMotor.getEncoder().getVelocity();
+    }
+
+    @AutoLogOutput(key = "upBeat/error")
+    private double getError(){
+        return setPoint-getSpeed();
+    }
+
+    private boolean atSpeed(){
+        if(getError()<-100 && getError()>100){
+        return true;
+        }
+        return false;
+    }
+
+    private void setSpeed(){
+        topPid.setReference((setPoint*1.2), ControlType.kVelocity);
+        bottomPid.setReference(setPoint, ControlType.kVelocity);
+    }
+
+    private Command speedCommand(double speed){
+        setPoint = speed;
+        return run (()-> setSpeed()).until(this::atSpeed); 
+    }
+
     public Command shootNote() {
         return startEnd(
             () -> {
-                topPid.setReference(4800, ControlType.kVelocity);
-                bottomPid.setReference(4000, ControlType.kVelocity);
+                speedCommand(Constants.shoot);
             },
             () -> {
-                topPid.setReference(0, ControlType.kVelocity);
-                bottomPid.setReference(0, ControlType.kVelocity);
+                speedCommand(Constants.stop);
             }
             );
     }
```

Comment:
As a startEnd command, we'll just need to copy the lambda from our speedCommand into it.

That said, we'll want to rename this to something like `shootThenStop` so it can be used in teleop manual control and add a new `shootNote` command which returns a `speedCommand` like the others for auto.